src/stream/futures_unordered.rs

```diff
@@ -440,6 +451,37 @@ impl<F: Future> FromIterator<F> for FuturesUnordered<F> {
     }
 }
 
+#[derive(Debug)]
+/// Mutable iterator over all futures in the unordered set.
+pub struct IterMut<'a, F: 'a> {
+    node: *const Node<F>,
+    len: usize,
+    _marker: PhantomData<&'a F>
```

Comment:
I think this may best be `&'a mut FuturesUnordered<F>` perhaps?

Reply:
Apologies -- can you go into more details why that would be?

Reply:
Long long ago in Rust there were subtle distinctions between how `&mut T` and `&T` worked, and I forget if any of them are still true today. Basically because this is a mutable iterator it's "better" to have a phantom data of `&mut ...` instead of `&...` to ensure this information is conveyed to the compiler. 

Additionally I figure it's best to store `FuturesUnordered<F>` as that's effectively what's happening here (vs just `F`). This can also cause subtle differences but probably not as important as `&mut` vs not

Reply:
Thanks for explaining that, changed it!